helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -737,4 +843,100 @@ public HeliosDeployment build() {
       return new HeliosSoloDeployment(this);
     }
   }
+
+  private static class DefaultSoloMasterProber implements SoloMasterProber {
+
+    private static final Logger log = LoggerFactory.getLogger(DefaultSoloMasterProber.class);
+
+    private HostAndPort hostAndPort = null;
+
+    /**
+     * @return true if we can connect to the specified host and port. Null, otherwise.
+     * @throws Exception
+     */
+    @Override
+    public Boolean call() throws Exception {
+      if (hostAndPort == null) {
+        throw new IllegalArgumentException(
+            "HeliosMasterSocketProber.hostAndPort must be set first.");
+      }
+
+      final SocketAddress address = new InetSocketAddress(hostAndPort.getHostText(),
+                                                          hostAndPort.getPort());
+      log.debug("attempting to connect to {}", address);
+
+      try {
+        final Socket s = new Socket();
+        s.connect(address, 100);
```

Comment:
Would this be susceptible to the same issue described in `connectToWatchdog()`? i.e. Do we need to read/write some data to ensure it's really connected?